### PR TITLE
Updating EventAggregator sample

### PR DESCRIPTION
**Pull request by jbachelor**
The way this sample was, the `IsFun` property of the `DataEntryPageViewModel` defaulted to the false/off position no matter what the switch was set to on the `HomePageViewModel`. With these proposed changes, the current value of `IsFun` is passed to the `DataEntryPageViewModel` to reflect the current value, rather than always defaulting to false. 

I believe this makes the app more logical and predictable, now that the `IsFun` property of the `HomePageViewModel` will always reflect the value displayed on the `DataEntryPageViewModel`:

Old Behavior | New Behavior
------------- | ------
| <ol><li>Launch app</li><li>Tap "XAMARIN.FORMS EVENTS"</li><li>Tap "PROVIDE FEEDBACK"</li><li>Set the "Isn't this fun?" switch to true/on</li><li>Tap "SUBMIT"</li><li>Note switch in the true/on position now on `HomePage`</li><li>Tap "PROVIDE FEEDBACK" again</li><li>**Note switch on the `DataEntryPage` is in the false/off position, which does not reflect the value on the `HomePage`**</li></ol> | <ol><li>Launch app</li><li>Tap "XAMARIN.FORMS EVENTS"</li><li>Tap "PROVIDE FEEDBACK"</li><li>Set the "Isn't this fun?" switch to true/on</li><li>Tap "SUBMIT"</li><li>Note switch in the true/on position now on `HomePage`</li><li>Tap "PROVIDE FEEDBACK" again</li><li>**Note switch on the `DataEntryPage` is in the true/on position, accurately reflecting the value on the `HomePage` and allowing the user to set the switch to the false/off position which will update the `HomePage` value as well**</li></ol> |

### CODE REVIEW

File: 05-EventAggregator/src/PrismSample/ViewModels/DataEntryPageViewModel.cs [hunk 1]
```diff
@@ -34,5 +35,14 @@ private void OnSubmitTapped()
         {
             _navigationService.GoBackAsync ();
         }
+
+        public override void OnNavigatedTo(INavigationParameters parameters)
+        {
+            base.OnNavigatedTo(parameters);
```
Comment (dansiegel) [inline 05-EventAggregator/src/PrismSample/ViewModels/DataEntryPageViewModel.cs:41]: no need for the base.OnNavigatedTo

File: 05-EventAggregator/src/PrismSample/ViewModels/DataEntryPageViewModel.cs [hunk 2]
```diff
@@ -34,5 +35,14 @@ private void OnSubmitTapped()
...
+            if (parameters.ContainsKey(NavigationParameterKeys.CurrentIsFunValue))
+            {
+                IsFun = (bool)parameters[NavigationParameterKeys.CurrentIsFunValue];
```
Comment (dansiegel) [inline 05-EventAggregator/src/PrismSample/ViewModels/DataEntryPageViewModel.cs:44]: this is a deprecated method. Use `parameters.GetValue<bool>("CurrentIsFunValue");`

File: 05-EventAggregator/src/PrismSample/ViewModels/HomePageViewModel.cs [hunk 3]
```diff
@@ -48,7 +48,11 @@ public bool IsFun
 
         private void OnEntryCommandTapped()
         {
-            _navigationService.NavigateAsync(Navigate.DataEntry);
+            INavigationParameters parameters = new NavigationParameters
+            {
+                {NavigationParameterKeys.CurrentIsFunValue, IsFun}
+            };
+            _navigationService.NavigateAsync(Navigate.DataEntry, parameters);
```
Comment (dansiegel) [inline 05-EventAggregator/src/PrismSample/ViewModels/HomePageViewModel.cs:55]: simplify this...

```cs
_navigationService.NavigateAsync(Navigate.DataEntry, (NavigationParameterKeys.CurrentIsFunValue, IsFun));
```

**Comment by jbachelor** [comment]
Thanks, @dansiegel! I'll make these changes and let you know when it's ready for your review.

**Comment by jbachelor** [comment]
@dansiegel - I believe I've made all the changes you were asking for... Please let me know if anything looks wrong, or if you notice anything else you think should change.

Thanks for all your hard work!

**Comment by dansiegel** [comment]
Thanks for the PR

**Pull request merged**